<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Faster implementations of phyloseq functions • speedyseq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Faster implementations of phyloseq functions">
<meta property="og:description" content="Faster implementations of phyloseq functions.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">speedyseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.3.9008</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mikemc/speedyseq/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="speedyseq" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#speedyseq" class="anchor"></a>speedyseq</h1></div>
<!-- badges: start -->

<p>Speedyseq is an R package for microbiome data analysis that extends the popular <a href="https://joey711.github.io/phyloseq/">phyloseq</a> package. Speedyseq began with the limited goal of providing faster versions of phyloseq’s plotting and taxonomic merging functions, but now contains a growing number of enhancements to phyloseq which I have found useful.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Install the current development version with the remotes package,</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("remotes")</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"mikemc/speedyseq"</span><span class="op">)</span></code></pre></div>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<p>Method 1: Call speedyseq functions explicitly when you want to use speedyseq’s version instead of phyloseq. This method ensures that you do not unintentionally call speedyseq’s version of a phyloseq function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span>
  <span class="co"># Calls phyloseq's psmelt</span>
  <span class="va">df1</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/psmelt.html">psmelt</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span> <span class="co"># slow</span>
<span class="op">)</span>
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;   6.320   0.063   6.390</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span>
  <span class="va">df2</span> <span class="op">&lt;-</span> <span class="fu">speedyseq</span><span class="fu">::</span><span class="fu"><a href="reference/psmelt.html">psmelt</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span> <span class="co"># fast</span>
<span class="op">)</span>
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;   0.344   0.004   0.245</span>
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/all_equal.html">all_equal</a></span><span class="op">(</span><span class="va">df1</span>, <span class="va">df2</span>, ignore_row_order <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="kw"><a href="https://rdrr.io/r/base/detach.html">detach</a></span><span class="op">(</span><span class="va">package</span><span class="op">:</span><span class="va">phyloseq</span><span class="op">)</span></code></pre></div>
<p>Method 2: Load speedyseq, which will load phyloseq and all speedyseq functions and cause calls to the overlapping function names to go to speedyseq by default.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://mikemc.github.io/speedyseq">speedyseq</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: phyloseq</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'speedyseq'</span>
<span class="co">#&gt; The following objects are masked from 'package:phyloseq':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter_taxa, plot_bar, plot_heatmap, plot_tree, psmelt, tax_glom, tip_glom,</span>
<span class="co">#&gt;     transform_sample_counts</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span>
  <span class="va">ps1</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_glom.html">tax_glom</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, <span class="st">"Genus"</span><span class="op">)</span> <span class="co"># slow</span>
<span class="op">)</span>
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;  31.856   0.106  32.031</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span>
  <span class="co"># Calls speedyseq's tax_glom</span>
  <span class="va">ps2</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/tax_glom.html">tax_glom</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, <span class="st">"Genus"</span><span class="op">)</span> <span class="co"># fast</span>
<span class="op">)</span>
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;   0.241   0.000   0.230</span></code></pre></div>
<p>Loading speedyseq will also load the <a href="https://magrittr.tidyverse.org/">magrittr</a> pipe (<code><a href="reference/%&gt;%.html">%&gt;%</a></code>) to allow pipe chains with phyloseq objects,</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gp.filt.prop</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="reference/filter_taxa.html">filter_taxa2</a></span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">.</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="reference/transform_sample_counts.html">transform_sample_counts</a></span><span class="op">(</span><span class="op">~</span> <span class="va">.</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="features" class="section level2">
<h2 class="hasAnchor">
<a href="#features" class="anchor"></a>Features</h2>
<div id="faster-implementations-of-phyloseq-functions" class="section level3">
<h3 class="hasAnchor">
<a href="#faster-implementations-of-phyloseq-functions" class="anchor"></a>Faster implementations of phyloseq functions</h3>
<ul>
<li>
<code><a href="reference/psmelt.html">psmelt()</a></code> and the plotting functions that use it: <code><a href="reference/plot_bar.html">plot_bar()</a></code>, <code><a href="reference/plot_heatmap.html">plot_heatmap()</a></code>, and <code><a href="reference/plot_tree.html">plot_tree()</a></code>.</li>
<li>The taxonomic merging functions <code><a href="reference/tax_glom.html">tax_glom()</a></code> and <code><a href="reference/tip_glom.html">tip_glom()</a></code>. Speedyseq’s <code><a href="reference/tip_glom.html">tip_glom()</a></code> also has significantly lower memory usage.</li>
</ul>
<p>These functions should generally function as drop-in replacements for phyloseq’s versions, with additional arguments allowing for modified behavior. Differences in row order (for <code><a href="reference/psmelt.html">psmelt()</a></code>) and taxon order (for <code><a href="reference/tax_glom.html">tax_glom()</a></code>) can occur; see <a href="https://mikemc.github.io/speedyseq/news/index.html">Changelog</a> for details.</p>
</div>
<div id="new-taxonomic-merging-functions" class="section level3">
<h3 class="hasAnchor">
<a href="#new-taxonomic-merging-functions" class="anchor"></a>New taxonomic merging functions</h3>
<ul>
<li>A general-purpose merging function <code><a href="reference/merge_taxa_vec.html">merge_taxa_vec()</a></code> that provides a vectorized version of phyloseq’s <code><a href="https://rdrr.io/pkg/phyloseq/man/merge_taxa-methods.html">merge_taxa()</a></code> function.</li>
<li>A function <code><a href="reference/tree_glom.html">tree_glom()</a></code> that performs direct phylogenetic merging of taxa. This function provides an alternative to the indirect phylogenetic merging done by <code><a href="reference/tip_glom.html">tip_glom()</a></code> that is much faster and arguably more intuitive.</li>
</ul>
<p>See the <a href="https://mikemc.github.io/speedyseq/news/index.html">Changelog</a> for details and examples.</p>
</div>
<div id="enhancements-and-additions-to-other-phyloseq-functions" class="section level3">
<h3 class="hasAnchor">
<a href="#enhancements-and-additions-to-other-phyloseq-functions" class="anchor"></a>Enhancements and additions to other phyloseq functions</h3>
<p>See the <a href="https://mikemc.github.io/speedyseq/reference/index.html">online documentation</a> for an up-to-date list and usage information and the <a href="https://mikemc.github.io/speedyseq/news/index.html">Changelog</a> for further information.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/mikemc/speedyseq/">https://​github.com/​mikemc/​speedyseq/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/mikemc/speedyseq/issues">https://​github.com/​mikemc/​speedyseq/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/AGPL-3">AGPL-3</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Michael McLaren <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://zenodo.org/badge/latestdoi/179732395"><img src="https://zenodo.org/badge/179732395.svg" alt="DOI"></a></li>
<li><a href="https://travis-ci.com/mikemc/speedyseq"><img src="https://travis-ci.com/mikemc/speedyseq.svg?branch=main" alt="Travis build status"></a></li>
<li><a href="https://codecov.io/gh/mikemc/speedyseq?branch=main"><img src="https://codecov.io/gh/mikemc/speedyseq/branch/main/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Michael McLaren.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
