<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot a phylogenetic tree with optional annotations — plot_tree • speedyseq</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Plot a phylogenetic tree with optional annotations — plot_tree" />
<meta property="og:description" content="There are many useful examples of phyloseq tree graphics in the
phyloseq online
tutorials. This function is intended to facilitate easy graphical
investigation of the phylogenetic tree, as well as sample data. Note that
for phylogenetic sequencing of samples with large richness, some of the
options in this function will be prohibitively slow to render, or too dense
to be interpretable. A rough ``rule of thumb'' is to use subsets of data
with not many more than 200 OTUs per plot, sometimes less depending on the
complexity of the additional annotations being mapped to the tree. It is
usually possible to create an unreadable, uninterpretable tree with modern
datasets. However, the goal should be toward parameter settings and data
subsets that convey (honestly, accurately) some biologically relevant
feature of the data. One of the goals of the phyloseq-package
is to make the determination of these features/settings as easy as possible." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">speedyseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.3.9018</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mikemc/speedyseq/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot a phylogenetic tree with optional annotations</h1>
    <small class="dont-index">Source: <a href='https://github.com/mikemc/speedyseq/blob/master/R/plot-methods.R'><code>R/plot-methods.R</code></a></small>
    <div class="hidden name"><code>plot_tree.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>There are many useful examples of phyloseq tree graphics in the
<a href='http://joey711.github.io/phyloseq/plot_tree-examples'>phyloseq online
tutorials</a>. This function is intended to facilitate easy graphical
investigation of the phylogenetic tree, as well as sample data. Note that
for phylogenetic sequencing of samples with large richness, some of the
options in this function will be prohibitively slow to render, or too dense
to be interpretable. A rough ``rule of thumb'' is to use subsets of data
with not many more than 200 OTUs per plot, sometimes less depending on the
complexity of the additional annotations being mapped to the tree. It is
usually possible to create an unreadable, uninterpretable tree with modern
datasets. However, the goal should be toward parameter settings and data
subsets that convey (honestly, accurately) some biologically relevant
feature of the data. One of the goals of the <code><a href='https://rdrr.io/pkg/phyloseq/man/phyloseq-package.html'>phyloseq-package</a></code>
is to make the determination of these features/settings as easy as possible.</p>
    </div>

    <pre class="usage"><span class='fu'>plot_tree</span><span class='op'>(</span>
  <span class='va'>physeq</span>,
  method <span class='op'>=</span> <span class='st'>"sampledodge"</span>,
  nodelabf <span class='op'>=</span> <span class='cn'>NULL</span>,
  color <span class='op'>=</span> <span class='cn'>NULL</span>,
  shape <span class='op'>=</span> <span class='cn'>NULL</span>,
  size <span class='op'>=</span> <span class='cn'>NULL</span>,
  min.abundance <span class='op'>=</span> <span class='cn'>Inf</span>,
  label.tips <span class='op'>=</span> <span class='cn'>NULL</span>,
  text.size <span class='op'>=</span> <span class='cn'>NULL</span>,
  sizebase <span class='op'>=</span> <span class='fl'>5</span>,
  base.spacing <span class='op'>=</span> <span class='fl'>0.02</span>,
  ladderize <span class='op'>=</span> <span class='cn'>FALSE</span>,
  plot.margin <span class='op'>=</span> <span class='fl'>0.2</span>,
  title <span class='op'>=</span> <span class='cn'>NULL</span>,
  treetheme <span class='op'>=</span> <span class='cn'>NULL</span>,
  justify <span class='op'>=</span> <span class='st'>"jagged"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>physeq</th>
      <td><p>(Required). The data about which you want to
plot and annotate a phylogenetic tree, in the form of a single instance of
the <code><a href='https://rdrr.io/pkg/phyloseq/man/phyloseq-class.html'>phyloseq-class</a></code>, containing at minimum a phylogenetic
tree component (try <code><a href='https://rdrr.io/pkg/phyloseq/man/phy_tree-methods.html'>phy_tree</a></code>).  One of the major advantages
of this function over basic tree-plotting utilities in the
<code>ape</code>-package is the ability to easily annotate the tree with
sample variables and taxonomic information. For these uses, the
<code>physeq</code> argument should also have a <code><a href='https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html'>sample_data</a></code> and/or
<code><a href='https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html'>tax_table</a></code> component(s).</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>(Optional). Character string. Default <code>"sampledodge"</code>.
The name of the annotation method to use.  This will be expanded in future
versions.  Currently only <code>"sampledodge"</code> and <code>"treeonly"</code> are
supported.  The <code>"sampledodge"</code> option results in points drawn next
to leaves if individuals from that taxa were observed, and a separate
point is drawn for each sample.</p></td>
    </tr>
    <tr>
      <th>nodelabf</th>
      <td><p>(Optional). A function. Default <code>NULL</code>.
If <code>NULL</code>, the default, a function will be selected for you based
upon whether or not there are node labels in <code><a href='https://rdrr.io/pkg/phyloseq/man/phy_tree-methods.html'>phy_tree(physeq)</a></code>.  For
convenience, the phyloseq package includes two generator functions for
adding arbitrary node labels (can be any character string),
<code><a href='https://rdrr.io/pkg/phyloseq/man/nodeplotdefault.html'>nodeplotdefault</a></code>; as well as for adding bootstrap values in a
certain range, <code><a href='https://rdrr.io/pkg/phyloseq/man/nodeplotboot.html'>nodeplotboot</a></code>.  To not have any node labels in
the graphic, set this argument to <code><a href='https://rdrr.io/pkg/phyloseq/man/nodeplotblank.html'>nodeplotblank</a></code>.</p></td>
    </tr>
    <tr>
      <th>color</th>
      <td><p>(Optional). Character string. Default <code>NULL</code>.
The name of the variable in <code>physeq</code> to map to point color.
Supported options here also include the reserved special variables of
<code><a href='psmelt.html'>psmelt</a></code>.</p></td>
    </tr>
    <tr>
      <th>shape</th>
      <td><p>(Optional). Character string. Default <code>NULL</code>.
The name of the variable in <code>physeq</code> to map to point shape.
Supported options here also include the reserved special variables of
<code><a href='psmelt.html'>psmelt</a></code>.</p></td>
    </tr>
    <tr>
      <th>size</th>
      <td><p>(Optional). Character string. Default <code>NULL</code>.  The name of
the variable in <code>physeq</code> to map to point size.  A special argument
<code>"abundance"</code> is reserved here and scales point size using abundance
in each sample on a log scale.  Supported options here also include the
reserved special variables of <code><a href='psmelt.html'>psmelt</a></code>.</p></td>
    </tr>
    <tr>
      <th>min.abundance</th>
      <td><p>(Optional). Numeric.  The minimum number of individuals
required to label a point with the precise number.  Default is <code>Inf</code>,
meaning that no points will have their abundance labeled.  If a vector,
only the first element is used.</p></td>
    </tr>
    <tr>
      <th>label.tips</th>
      <td><p>(Optional). Character string. Default is <code>NULL</code>,
indicating that no tip labels will be printed.  If <code>"taxa_names"</code>,
then the name of the taxa will be added to the tree; either next to the
leaves, or next to the set of points that label the leaves. Alternatively,
if this is one of the rank names (from <code><a href='https://rdrr.io/pkg/phyloseq/man/rank_names.html'>rank_names(physeq)</a></code>), then
the identity (if any) for that particular taxonomic rank is printed
instead.</p></td>
    </tr>
    <tr>
      <th>text.size</th>
      <td><p>(Optional). Numeric. Should be positive. The
size parameter used to control the text size of taxa labels.  Default is
<code>NULL</code>. If left <code>NULL</code>, this function will automatically
calculate a (hopefully) optimal text size given the vertical constraints
posed by the tree itself.  This argument is included in case the
automatically-calculated size is wrong, and you want to change it.  Note
that this parameter is only meaningful if <code>label.tips</code> is not
<code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>sizebase</th>
      <td><p>(Optional). Numeric. Should be positive.
The base of the logarithm used to scale point sizes to graphically
represent abundance of species in a given sample. Default is 5.</p></td>
    </tr>
    <tr>
      <th>base.spacing</th>
      <td><p>(Optional). Numeric. Default is <code>0.02</code>.
Should be positive.  This defines the base-spacing between points at each
tip/leaf in the the tree. The larger this value, the larger the spacing
between points.  This is useful if you have problems with overlapping
large points and/or text indicating abundance, for example. Similarly, if
you don't have this problem and want tighter point-spacing, you can shrink
this value.</p></td>
    </tr>
    <tr>
      <th>ladderize</th>
      <td><p>(Optional). Boolean or character string (either
<code>FALSE</code>, <code>TRUE</code>, or <code>"left"</code>).  Default is <code>FALSE</code>.
This parameter specifies whether or not to <code><a href='https://rdrr.io/pkg/ape/man/ladderize.html'>ladderize</a></code>
the tree (i.e., reorder nodes according to the depth of their enclosed
subtrees) prior to plotting.  This tends to make trees more aesthetically
pleasing and legible in a graphical display.  When <code>TRUE</code> or
<code>"right"</code>, <code>right'' ladderization is used.  When set to \code{FALSE}, no ladderization is applied.  When set to \code{"left"}, the reverse direction (</code>left'' ladderization) is applied.  This argument is
passed on to <code><a href='https://rdrr.io/pkg/phyloseq/man/tree_layout.html'>tree_layout</a></code>.</p></td>
    </tr>
    <tr>
      <th>plot.margin</th>
      <td><p>(Optional). Numeric. Default is <code>0.2</code>.  Should be
positive.  This defines how much right-hand padding to add to the tree
plot, which can be required to not truncate tip labels. The margin value
is specified as a fraction of the overall tree width which is added to the
right side of the plot area. So a value of <code>0.2</code> adds twenty percent
extra space to the right-hand side of the plot.</p></td>
    </tr>
    <tr>
      <th>title</th>
      <td><p>(Optional). Default <code>NULL</code>. Character string. The main
title for the graphic.</p></td>
    </tr>
    <tr>
      <th>treetheme</th>
      <td><p>(Optional). A custom <code>ggplot</code>2
<code><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></code> layer to use for the tree. Supplants any
default theme layers used within the function.  A value of <code>NULL</code>
uses a default, minimal-annotations theme.  If anything other than a them
or <code>NULL</code>, the current global ggplot2 theme will result.</p></td>
    </tr>
    <tr>
      <th>justify</th>
      <td><p>(Optional). A character string indicating the type of
justification to use on dodged points and tip labels.  A value of
<code>"jagged"</code>, the default, results in these tip-mapped elements being
spaced as close to the tips as possible without gaps.  Currently, any
other value for <code>justify</code> results in a left-justified arrangement of
both labels and points.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>ggplot</code>2 plot.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function received an early development contribution from the work of
Gregory Jordan via <a href='https://github.com/gjuggler/ggphylo'>the ggphylo
package</a>.  <code>plot_tree</code> has since been re-written.  For details see
<code><a href='https://rdrr.io/pkg/phyloseq/man/tree_layout.html'>tree_layout</a></code>.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code>plot.phylo</code></p>
<p>There are many useful examples of phyloseq tree graphics in the
<a href='http://joey711.github.io/phyloseq/plot_tree-examples'>phyloseq online
tutorials</a>.</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Using plot_tree() with the esophagus dataset.</span>
<span class='co'># Please note that many more interesting examples are shown in the online</span>
<span class='co'># tutorials at http://joey711.github.io/phyloseq/plot_tree-examples</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>esophagus</span><span class='op'>)</span>
<span class='co'># plot_tree(esophagus)</span>
<span class='co'># plot_tree(esophagus, color="Sample")</span>
<span class='co'># plot_tree(esophagus, size="Abundance")</span>
<span class='co'># plot_tree(esophagus, size="Abundance", color="samples")</span>
<span class='fu'>plot_tree</span><span class='op'>(</span><span class='va'>esophagus</span>, size<span class='op'>=</span><span class='st'>"Abundance"</span>, color<span class='op'>=</span><span class='st'>"Sample"</span>, base.spacing<span class='op'>=</span><span class='fl'>0.03</span><span class='op'>)</span>
</div><div class='img'><img src='plot_tree-1.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'>plot_tree</span><span class='op'>(</span><span class='va'>esophagus</span>, size<span class='op'>=</span><span class='st'>"abundance"</span>, color<span class='op'>=</span><span class='st'>"samples"</span>, base.spacing<span class='op'>=</span><span class='fl'>0.03</span><span class='op'>)</span>
</div><div class='img'><img src='plot_tree-2.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Michael McLaren.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


