<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Melt phyloseq data object into large data.frame — psmelt • speedyseq</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Melt phyloseq data object into large data.frame — psmelt" />
<meta property="og:description" content="The psmelt function is a specialized melt function for melting phyloseq
objects (instances of the phyloseq class), usually for producing graphics
with ggplot2::ggplot2. The naming conventions used in downstream phyloseq
graphics functions have reserved the following variable names that should
not be used as the names of sample_variables() or taxonomic
rank_names(). These reserved names are c(&quot;Sample&quot;, &quot;Abundance&quot;, &quot;OTU&quot;).
Also, you should not have identical names for sample variables and taxonomic
ranks. That is, the intersection of the output of the following two
functions sample_variables(), rank_names() should be an empty vector
(e.g.  intersect(sample_variables(physeq), rank_names(physeq))). All of
these potential name collisions are checked-for and renamed automatically
with a warning. However, if you (re)name your variables accordingly ahead of
time, it will reduce confusion and eliminate the warnings." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">speedyseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.3.9018</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mikemc/speedyseq/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Melt phyloseq data object into large data.frame</h1>
    <small class="dont-index">Source: <a href='https://github.com/mikemc/speedyseq/blob/master/R/psmelt.R'><code>R/psmelt.R</code></a></small>
    <div class="hidden name"><code>psmelt.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The psmelt function is a specialized melt function for melting phyloseq
objects (instances of the phyloseq class), usually for producing graphics
with <a href='https://ggplot2.tidyverse.org/reference/ggplot2-package.html'>ggplot2::ggplot2</a>. The naming conventions used in downstream phyloseq
graphics functions have reserved the following variable names that should
not be used as the names of <code><a href='https://rdrr.io/pkg/phyloseq/man/sample_variables.html'>sample_variables()</a></code> or taxonomic
<code><a href='https://rdrr.io/pkg/phyloseq/man/rank_names.html'>rank_names()</a></code>. These reserved names are <code><a href='https://rdrr.io/r/base/c.html'>c("Sample", "Abundance", "OTU")</a></code>.
Also, you should not have identical names for sample variables and taxonomic
ranks. That is, the intersection of the output of the following two
functions <code><a href='https://rdrr.io/pkg/phyloseq/man/sample_variables.html'>sample_variables()</a></code>, <code><a href='https://rdrr.io/pkg/phyloseq/man/rank_names.html'>rank_names()</a></code> should be an empty vector
(e.g.  <code><a href='https://rdrr.io/r/base/sets.html'>intersect(sample_variables(physeq), rank_names(physeq))</a></code>). All of
these potential name collisions are checked-for and renamed automatically
with a warning. However, if you (re)name your variables accordingly ahead of
time, it will reduce confusion and eliminate the warnings.</p>
    </div>

    <pre class="usage"><span class='fu'>psmelt</span><span class='op'>(</span><span class='va'>physeq</span>, as <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"speedyseq.psmelt_class"</span><span class='op'>)</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>physeq</th>
      <td><p>An <a href='https://rdrr.io/pkg/phyloseq/man/otu_table-class.html'>otu_table</a> or <a href='https://rdrr.io/pkg/phyloseq/man/phyloseq-class.html'>phyloseq</a>; most useful for
phyloseq-class.</p></td>
    </tr>
    <tr>
      <th>as</th>
      <td><p>Class of the output table; see Details.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A table of the specified class</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The <code>as</code> argument allows choosing between three classes for tabular data:</p><ul>
<li><p>data.frame is chosen by "data.frame" or "df"</p></li>
<li><p>data.table is chosen by "data.table" or "dt"</p></li>
<li><p><a href='https://tibble.tidyverse.org/reference/tbl_df-class.html'>tbl_df</a> is chosen by "tbl_df", "tbl", or "tibble"</p></li>
</ul>

<p>The default is "data.frame" and can be overridden by setting the
"speedyseq.psmelt_class" option.</p>
<p>Note that "melted" phyloseq data is stored much less efficiently, and so RAM
storage issues could arise with a smaller dataset (smaller number of
samples/OTUs/variables) than one might otherwise expect.  For common sizes
of graphics-ready datasets, however, this should not be a problem.  Because
the number of OTU entries has a large effect on the RAM requirement, methods
to reduce the number of separate OTU entries -- for instance by
agglomerating OTUs based on phylogenetic distance using <code><a href='tip_glom.html'>tip_glom()</a></code> -- can
help alleviate RAM usage problems. This function is made user-accessible for
flexibility, but is also used extensively by plot functions in phyloseq.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='plot_bar.html'>plot_bar()</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>"GlobalPatterns"</span><span class='op'>)</span>
<span class='va'>gp.ch</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html'>subset_taxa</a></span><span class='op'>(</span><span class='va'>GlobalPatterns</span>, <span class='va'>Phylum</span> <span class='op'>==</span> <span class='st'>"Chlamydiae"</span><span class='op'>)</span>
<span class='va'>mdf</span> <span class='op'>=</span> <span class='fu'>psmelt</span><span class='op'>(</span><span class='va'>gp.ch</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>mdf</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] 546</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>mdf</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] 17</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>mdf</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "OTU"                      "Sample"                  
#&gt;  [3] "Abundance"                "X.SampleID"              
#&gt;  [5] "Primer"                   "Final_Barcode"           
#&gt;  [7] "Barcode_truncated_plus_T" "Barcode_full_length"     
#&gt;  [9] "SampleType"               "Description"             
#&gt; [11] "Kingdom"                  "Phylum"                  
#&gt; [13] "Class"                    "Order"                   
#&gt; [15] "Family"                   "Genus"                   
#&gt; [17] "Species"                 </div><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://tibble.tidyverse.org/reference/rownames.html'>rownames</a></span><span class='op'>(</span><span class='va'>mdf</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "436" "243" "84"  "417" "395" "458"</div><div class='input'><span class='co'># Create a ggplot similar to</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://ggplot2.tidyverse.org'>"ggplot2"</a></span><span class='op'>)</span>
<span class='va'>p</span> <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='va'>mdf</span>, <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x<span class='op'>=</span><span class='va'>SampleType</span>, y<span class='op'>=</span><span class='va'>Abundance</span>, fill<span class='op'>=</span><span class='va'>Genus</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>p</span> <span class='op'>=</span> <span class='va'>p</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bar.html'>geom_bar</a></span><span class='op'>(</span>color<span class='op'>=</span><span class='st'>"black"</span>, stat<span class='op'>=</span><span class='st'>"identity"</span>, position<span class='op'>=</span><span class='st'>"stack"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>p</span><span class='op'>)</span>
</div><div class='img'><img src='psmelt-1.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Michael McLaren.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


